src/components/common/Map/index.js

```diff
@@ -20,6 +27,22 @@ const UpdateCenter = ({ position }) => {
 const Map = ({ position }) => {
   const history = useHistory();
   const { hasNewTarget, newTarget } = useNewTarget();
+  const topics = useSelector(state => state.targetReducer.topics, shallowEqual);
+  const targets = useSelector(state => state.targetReducer.targets, shallowEqual);
+  const getTopics = useDispatch(getTargetTopics);
+  const getTargets = useDispatch(getTargetsAction);
+
+  useEffect(() => {
+    if (topics.length === 0) {
+      getTopics();
+    }
+  }, [topics, getTopics]);
+
+  useEffect(() => {
+    if (targets.length === 0) {
+      getTargets();
+    }
+  }, [getTargets, targets.length]);
```

Comment:
Isn't the result of this running it only on mount? if it is, it can just be replaced by this:
```suggestion
  useEffect(() => {
    getTopics();
    getTargets();
  }, []);
```

Reply:
Yes! I think the only case in which this code has a different behavior is when another component had already called either getTopics or getTargets, and so another call to the backend would be unnecessary. Is this right or am I losing something in the way? 😄

In the next PR: https://github.com/nachosoloaga/target-mvd/pull/11/files# I've already moved those calls to a higher component this way.